gitea/resource_gitea_token.go

```
@@ -0,0 +1,79 @@
+package gitea
+
+import (
+	"fmt"
+	"code.gitea.io/sdk/gitea"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTokenCreate(d *schema.ResourceData, meta interface{}) (err error) {
+
+	client := meta.(*gitea.Client)
+
+	var opt gitea.CreateAccessTokenOption
+        opt.Name = d.Get("name").(string)
+
+        token, _, err := client.CreateAccessToken(opt)
+
+	if true || err == nil {
+		err = setTokenResourceData(token, d)
+	}
+	return
+}
+
+func resourceTokenRead(d *schema.ResourceData, meta interface{}) (err error) {
+	return
+}
+
+func resourceTokenDelete(d *schema.ResourceData, meta interface{}) (err error) {
+
+	client := meta.(*gitea.Client)
+
+	_, err = client.DeleteAccessToken(d.Get("name"))
```

i would argue that error handling here would make sense. else the provider would report a access token as deleted while in reality it just silently failed